src/App.tsx

```diff
@@ -1,24 +1,204 @@
+
+  const loadTodosData = async () => {
+    try {
+      const todosFromServer = await getTodos(USER_ID);
+
+      setTempTodo(null);
+      setInputDisable(false);
+      setTodos(todosFromServer);
+    } catch (error) {
+      setErrorMessage("Can't load data...");
+    }
+  };
+
+  useEffect(() => {
+    loadTodosData();
+  }, []);
+
+  const createNewTodo = (query: string) => {
+    const newId = Math.max(...todos.map((todo) => todo.id + 1));
+
+    const newTodo = {
+      id: newId,
+      userId: USER_ID,
+      title: query,
+      completed: false,
+    };
+
+    return newTodo;
+  };
+
+  const onCompletedChange = async (todo: Todo) => {
+    try {
+      await onUpdate(todo.id, {
+        id: todo.id,
+        userId: todo.userId,
+        title: todo.title,
+        completed: !todo.completed,
+      });
+
+      setTempTodo(null);
+      loadTodosData();
+    } catch (error) {
+      setErrorMessage('Unable to update a todo');
+    }
+  };
+
+  const dataTodo = {
+    userId: USER_ID,
+    title,
+    completed: false,
+  };
+
+  const sendTodo = async () => {
+    setTempTodo({ id: 0, ...dataTodo });
+
+    try {
+      if (title) {
+        await addTodo(createNewTodo(title));
+      } else {
+        setErrorMessage('Empty title alowed');
+      }
+
+      loadTodosData();
+    } catch {
+      setTempTodo(null);
+      setErrorMessage('Unable to add a todo');
+      setInputDisable(false);
+    }
+  };
+
+  const handleFormSubmit = (event: React.FormEvent) => {
+    setInputDisable(true);
+
+    event.preventDefault();
+
+    sendTodo();
+    setTitle('');
+  };
+
+  const removeTodo = async (todoId: number) => {
+    try {
+      await onDelete(todoId);
+      loadTodosData();
+    } catch (error) {
+      setTempTodo(null);
+      setErrorMessage('Unable to delete a todo');
+    }
+  };
+
+  const toggleAllComlpleted = async () => {
+    visibleTodos.map((todo) => {
+      return onCompletedChange(todo);
+    });
+  };
+
+  const clearCompleted = () => {
+    const clearCompletedTodos = todos.filter((todo) => todo.completed);
+
+    clearCompletedTodos.forEach((todo) => removeTodo(todo.id));
+  };
+
   if (!USER_ID) {
     return <UserWarning />;
   }
 
   return (
-    <section className="section container">
-      <p className="title is-4">
-        Copy all you need from the prev task:
-        <br />
-        <a href="https://github.com/mate-academy/react_todo-app-add-and-delete#react-todo-app-add-and-delete">React Todo App - Add and Delete</a>
-      </p>
-
-      <p className="subtitle">Styles are already copied</p>
-    </section>
+    <div className="todoapp">
+      <h1 className="todoapp__title">todos</h1>
+
+      <div className="todoapp__content">
+        <Header
+          title={title}
+          setTitle={setTitle}
+          onSubmit={handleFormSubmit}
+          isDisabled={inputDisable}
+          toggleAll={toggleAllComlpleted}
+          noCompletedTodos={hasCompleteTodos}
```

Comment:
why **no**CompleatedTodos if **has**Compleate(ed)Todos?

but anyway, you should know if the all todos are completed:
![image](https://user-images.githubusercontent.com/86923089/224533359-48d8f19c-b914-4d1a-88c8-5ae0e56062e2.png)